src/python/pants/backend/python/docs/sphinx/rules.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
I'm confused. Where is all the code? ;-)

It's pretty cool that there is substantially more test code and boilerplate than implementation code.